WordPress/src/main/java/org/wordpress/android/ui/posts/PostsPagerAdapter.kt

```diff
@@ -0,0 +1,38 @@
+package org.wordpress.android.ui.posts
+
+import android.content.Context
+import android.support.v4.app.FragmentManager
+import android.support.v4.app.FragmentPagerAdapter
+import android.view.ViewGroup
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.ui.posts.PostListType.DRAFTS
+import org.wordpress.android.ui.posts.PostListType.PUBLISHED
+import org.wordpress.android.ui.posts.PostListType.SCHEDULED
+import org.wordpress.android.ui.posts.PostListType.TRASHED
+import java.lang.ref.WeakReference
+
+class PostsPagerAdapter(private val site: SiteModel, val context: Context, val fm: FragmentManager) :
```

Comment:
This looks _very_ similar to `PagesPagerAdapter`, we should consider converting both to a reusable component in a later PR.

Reply:
I just realized this while reviewing #9233. Could we get away with not passing the `Context`? I think we can just return the string resource from `getPageTitle`.

I really hate having `Context` everywhere so I'd love to avoid having a reference to it if it's possible.

Reply:
>I really hate having `Context` everywhere so I'd love to avoid having a reference to it if it's possible.
I agree, I try avoid passing context wherever possible. However, I'm not sure I follow. 

- The method's return parameter is `CharSequence` so we need to convert it from resId -> text.
- We can replace the activity context with Application context, however the FragmentManager contains a reference to the Activity context anyway.

Reply:
Fixed in [39a5ad4](https://github.com/wordpress-mobile/WordPress-Android/pull/9226/commits/39a5ad4ce6e3530c096b11a24ae5fcc5e804f5a4)

Reply:
Apparently I am blind because I didn't see the `override`, sorry :(